test/functional/rpc_net.py

```diff
@@ -371,11 +371,6 @@ def test_getaddrmaninfo(self):
         self.log.info("Test getaddrmaninfo")
         node = self.nodes[1]
 
-        self.log.debug("Test that getaddrmaninfo is a hidden RPC")
-        # It is hidden from general help, but its detailed help may be called directly.
-        assert "getaddrmaninfo" not in node.help()
-        assert "getaddrmaninfo" in node.help("getaddrmaninfo")
```

Comment:
is there a reason why we're removing this test and not changing it to test that it isn't a hidden rpc

Reply:
mostly because all the other functional tests only check whether it's a hidden rpc. i think we'd have coverage from doing something like `assert "getaddrmaninfo" in node.help()` already with [`write_all_rpc_commands`](https://github.com/bitcoin/bitcoin/blob/97f756b12c8d8a9d3b621f296725dd7bf36bc8a9/test/functional/test_framework/coverage.py#L81).

Reply:
IMO I'd leave the validation to detect whether the rpc command is hidden or not and make the change to the failing test accordingly depending if the change was intentional. 

I've checked [write_all_rpc_commands](https://github.com/bitcoin/bitcoin/blob/97f756b12c8d8a9d3b621f296725dd7bf36bc8a9/test/functional/test_framework/coverage.py#L81) and I had to run the `test_runner.py` with argument `--coverage` twice with and without this branch in order to see if a rpc command is written in the coverage file or not (`node.help()`).

I think leaving the test as is it's more practical, but perhaps I'm missing something or it's just not as important.

Reply:
After an offline chat, I do agree with the author that while we have these checks being performed to verify that an `rpc command` is hidden, the reverse action doesn't make much sense as it's not as dangerous.